### PR TITLE
[com_messages] Implement searchtools

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.
#### Summary of Changes

Like done for so many other views across Joomla backend, this PR adds the search tools to com_messages messages view.
Also improved mobile view, added a "All" filter in the message state and some minor adjustements.
###### Before Patch

image
###### After Patch

image
#### Testing Instructions
1. Apply patch
2. Go to "Components" -> "Messaging"
3. Send some Messages across users (you will need to create another user for that)
4. Check if the searchtools, filters and ordering are working properly.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 65cb116ad3d0924af8a19809a2b9583f2b105353<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9876">issues.joomla.org/joomla-cms/9876</a>.</sub>

File: administrator/language/en-GB/en-GB.com_messages.ini [hunk 1]
```diff
@@ -27,9 +27,17 @@ COM_MESSAGES_FIELD_SUBJECT_LABEL="Subject"
 COM_MESSAGES_FIELD_USER_ID_FROM_LABEL="From"
 COM_MESSAGES_FIELD_USER_ID_TO_DESC="You must select a recipient."
 COM_MESSAGES_FIELD_USER_ID_TO_LABEL="Recipient"
+COM_MESSAGES_FILTER_STATES_DESC="Filter by message state"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_messages.ini]: If it is a DESC then please add a . at the end to match the style guide

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_messages.ini]: Where _DESC is description not descending

**Comment by andrepereiradasilva** [comment]
done

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on dbafbcbb35f521b3778d775826943b4042fe4ca0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9876">issues.joomla.org/joomla-cms/9876</a>.</sub>

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on dbafbcbb35f521b3778d775826943b4042fe4ca0<br /><br />thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9876">issues.joomla.org/joomla-cms/9876</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@brianteeman @MATsxm thanks for testing.

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9876">issues.joomla.org/joomla-cms/9876</a>.</sub>

**Comment by rdeutz** [comment]
@andrepereiradasilva could you check the conflicts, thanks

**Comment by andrepereiradasilva** [comment]
ignore again the unit tests label, somehow my git changed the filename of the xcache unit test, i reverted. all good now and no conflicts.

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on 933080008e8518fc14a3c46b9e717813e1a62906<br /><br />I don't know if it needs  to be tested again but... it is!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9876">issues.joomla.org/joomla-cms/9876</a>.</sub>

**Pull request merged**